src/hyp3_autorift/process.py

```diff
@@ -538,6 +592,13 @@ def main():
     g1, g2 = sorted(args.granules, key=get_datetime)
 
     product_file, browse_file = process(g1, g2, parameter_file=args.parameter_file, naming_scheme=args.naming_scheme)
+    thumbnail_file = create_thumbnail(browse_file)
+
+    if args.publish:
+        prefix = get_opendata_prefix(product_file)
+        upload_file_to_s3(product_file, OPEN_DATA_BUCKET, prefix)
+        upload_file_to_s3(browse_file, OPEN_DATA_BUCKET, prefix)
+        upload_file_to_s3(thumbnail_file, OPEN_DATA_BUCKET, prefix)
```

Comment:
Thinking about it, I think I'd rather do that after we upload to the HyP3 content bucket -- that way, if this part breaks, we still have a copy in HyP3 and can copy the files over without having to reprocess. 

What do you thinK?

Reply:
very good idea. I will change the sequence.